test/acceptance/filepermissions_test.go

```diff
@@ -0,0 +1,114 @@
+package acceptance
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"testing"
+	"time"
+)
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+const (
+	agentRuntime         = 1 * time.Second
+	agentConfigLocalPath = "agent_configs/minimum_config.json"
+	agentConfigPath      = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	agentConfigCopiedDir = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d"
+	translatedTomlPath   = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml"
+)
+
+func TestFilePermissions(t *testing.T) {
+	common.CopyFile(agentConfigLocalPath, agentConfigPath)
+	err := common.StartAgent(agentConfigPath, false)
+	if err != nil {
+		log.Printf("Agent failed to start due to err=%v\n", err)
+	}
+	time.Sleep(agentRuntime)
+	common.StopAgent()
+	groupResult := createGroupResult()
+	for _, testResult := range groupResult.TestResults {
+		assert.Equal(t, testResult.Status, status.SUCCESSFUL)
+	}
+	err = common.DeleteFile(agentConfigPath)
+	if err != nil {
+		log.Printf("Failed to delete config file; err=%v\n", err)
+	}
+}
+
+func createGroupResult() status.TestGroupResult {
+	testResults := make([]status.TestResult, 3)
+	testGroupResult := status.TestGroupResult{
+		Name:        "FilePermissions",
+		TestResults: testResults,
+	}
+	testResults[0] = onlyUserCanWriteToFilepath("root", agentConfigPath)
+	testResults[1] = onlyUserCanWriteToFilepath("cwagent", agentConfigCopiedDir)
+	testResults[2] = onlyUserCanWriteToFilepath("root", translatedTomlPath)
```

Comment:
This test is failing for some reason. Somehow, when I refactored this test, the `toml` file owner changed from `root` to `cwagent`. I went through line by line and couldn't understand why this behavior changed